### DISCUSSION TITLE
How to phrase fetch URL in PUT request for production with process.cwd

### DISCUSSION THREAD

**Discussion by christiankozalla** [vercel/next.js#15271] [Help] [answered]
Hello everyone! I dont know how to exactly phrase the fetch URL in a PUT request to API endpoint that updates a SQLite Database.

In development I had this: 

```js
const updateColumn = async (id, column) => {
  const endpoint = `http://localhost:3000/api/posts/${id}`;

  try {
    await fetch(endpoint, {
      method: "PUT",
      headers: {
        "Content-type": "application/json; charset=UTF-8",
      },
      body: JSON.stringify({ column: column }),
    });
  } catch (error) {
    console.error(error);
  }
};
```

I tried using `process.cwd()` but it didn't work out.

I can get the json data manually here `https://nextjs-blog.christiankozalla.vercel.app/api/posts/first-next-js-blog-devdiary` (this is an example for one blog post)
But how can I formulate this inside fetch?

Big Thanks in Advance!
Christian

**Comment by mattcarlotta** [comment, marked as answer]
You can use [NEXT_PUBLIC](https://nextjs.org/docs/basic-features/environment-variables#exposing-environment-variables-to-the-browser) within an [env file](https://nextjs.org/docs/basic-features/environment-variables#default-environment-variables) to use different API endpoints based upon the environment.

By default, env variables are converted to strings, so they don't need quotes.

**env.development**
```NEXT_PUBLIC_API=http://localhost:3000/api```

**env.production**
```NEXT_PUBLIC_API=https://nextjs-blog.christiankozalla.vercel.app/api```

```js
const updateColumn = async (id, column) => {
  try {
    await fetch(`${process.env.NEXT_PUBLIC_API}/posts/${id}`, {
      method: "PUT",
      headers: {
        "Content-type": "application/json; charset=UTF-8",
      },
      body: JSON.stringify({ column: column }),
    });
  } catch (error) {
    console.error(error);
  }
};
```

**Reply by christiankozalla** [reply]
Thank you very much! That solves it. But I am afraid there is still something wrong with my request handler or database query. Weird thing is, everything works in development.. and I am not very experienced in programming yet. I'm going to open a new question for the request handler. Thanks again!
